command/agent.go

```diff
@@ -716,6 +716,9 @@ func (c *AgentCommand) Run(args []string) int {
 			// Create a muxer and add paths relevant for the lease cache layer
 			mux := http.NewServeMux()
 			mux.Handle(consts.AgentPathCacheClear, leaseCache.HandleCacheClear(ctx))
+			if lnConfig.AgentAPI != nil && lnConfig.AgentAPI.EnableQuit {
+				mux.Handle(consts.AgentPathQuit, c.handleQuit())
+			}
```

Comment:
I noticed while testing this that if enable_quit=false, requests to /agent/v1/quit are proxied through to Vault which returns a 404 (as you'd expect). Alternatively we could decide to handle or not inside the handleQuit() function, which means the request would always stop at agent and not be proxied to Vault and back.

(Sorry, just thinking out loud here. I don't have a strong preference one way or the other.)

Reply:
Wow good catch - yeah I've updated this so we always serve the endpoint, but just serve the 404 from Agent instead when it's not enabled.